### ISSUE TITLE
base-spells.yaml / combat-trainer.lic - Heal not being cast as a buff

### ISSUE THREAD

**Issue by mikejcook** [rpherbig/dr-scripts#5145]
With the metaspell Adaptive Curing, the Empath spell Heal may be cast as a buff - it will wait around without injuries to heal you when you are hurt. combat-trainer doesn't recognize the spell as a re-castable buff. I identified this filter as the fail point:

.select { |_name, data| data['recast'] || data['recast_every'] || data['expire'] }
 -- def check_buffs(game_state)

So, I added recast: 1 to the Heal spell in base-spells.yaml, but I'm not really sure what recast: 1 means, other than the fact that it seems to be set on other buffs. Is it just 1 = true?

The change has fixed combat-trainer so that it recasts the spell, but I'm hesitant to create a pull request when I'm not positive what the change I did means, just because it fixed the issue.

**Comment by vtcifer** [comment]
IMO unless this change is non-breaking for characters without ADC, it should be managed in users yaml instead of base data.

**Comment by wstampley** [comment]
Yeah, the base effect of heal is that it heals wounds you currently have. If you don't have ADC, then heal ends when you have no wounds left and the entry having a 'recast' value means that it would get recast over and over for someone without wounds.

The right answer is to set heal to `recast: -1` for people who have ADC, in their YAML. This tells the magic system in Lich to only cast it if it isn't active, while letting you run it as a buff. Probably worth documenting... somewhere? SampleEmpath-setup maybe?

**Comment by mikejcook** [comment]
Yeah, that seems reasonable. I just checked and it does work as expected when I override in my YAML, so that seems good to go.

We should document it on this page, as it's a big part of what I've used to get my Empath's YAML into a usable state. If it had been there, I would've made the change and been good to go. The page could use a little updating as a whole, so I'll take a pass at clarifying the Empath Self Healing section (HW and HS aren't actually supported) and add a note about using Heal as a buff.
https://github.com/rpherbig/dr-scripts/wiki/Empath-Tutorials

Oh, and it looks like the recast IS documented in the sample YAML, I was just relying on the tutorial page. So we should be good to close this out.
